### PR TITLE
Frontnew

### PR DESCRIPTION
**Pull request by mechupieniach**

### CODE REVIEW

**Review by AltoDomino — APPROVED** [review]
jest w pyte

File: src/App.jsx [hunk 1]
```diff
@@ -3,18 +3,20 @@ import React, {useState, useEffect} from "react";
...
+    const [responseData, setResponseData] = useState([])
 
-    useEffect(async () => {
-        const response = await fetch('http://localhost:8008/albums')
-        const data = await response.json()
-        setResponseData(data)
-    })
+    useEffect(() => {
+        fetch(`http://localhost:8008/value=${value}`)
+        .then(response => response.json())
```
Comment (FutureInventor) [inline src/App.jsx:10]: Why 'then'? Use await instead.

File: src/App.jsx [hunk 2]
```diff
@@ -3,18 +3,20 @@ import React, {useState, useEffect} from "react";
...
+        .then(data => setResponseData(data))
+    }, [value])
 
     return (
         <div>
             <input value={value} onChange={(e) => setValue(e.target.value)}></input>
-            <div>{responseData}</div>
+            <div>{responseData.map((result) => {
```
Comment (FutureInventor) [inline src/App.jsx]:
```suggestion
            <div>{responseData.map((result) => <div>{result.title}</div>)}</div>
```

**Comment by FutureInventor** [comment]
On the first render. The page should show all records.

File: src/App.jsx (hunk 1 above)
Comment (mechupieniach) [reply src/App.jsx:10]: Because I get an error, that you can't use async function i useEffect :)

Comment (mechupieniach) [reply src/App.jsx:10]: *in useEffect

Comment (FutureInventor) [reply src/App.jsx:10]: Alright

**Review by FutureInventor — DISMISSED** [review]
Great job 💯

File: src/App.jsx [hunk 3]
```diff
@@ -3,18 +3,19 @@ import React, {useState, useEffect} from "react";
...
+        .then(response => response.json())
+        .then(data => setResponseData(data))
+    }, [value])
 
     return (
         <div>
             <input value={value} onChange={(e) => setValue(e.target.value)}></input>
-            <div>{responseData}</div>
+            <div>{responseData.map((result) => <div>{result.title}</div>)}</div>
+                return <div>{result.title}</div>
```
Comment (FutureInventor) [inline src/App.jsx]: What is this 'return' for?

Comment (FutureInventor) [reply src/App.jsx]: Also, 18th line is broking the whole application.

**Comment by FutureInventor** [comment]
There is an error in the console 
image

File: src/App.jsx (hunk 3 above)
Comment (mechupieniach) [reply src/App.jsx]: Ok, I'm stupid. I accepted suggestion and didn't realise it will leave both options.

**Comment by mechupieniach** [comment]
Okay, I fixed those errors :)

**Review by FutureInventor — APPROVED** [review]
As we spoke on the chat, the request should not send `title=${value}` when the value is empty. So the fetch should look like this: `fetch('http://localhost:8008/albums${value && '?title=${value}'}')` and then on the BE `url.includes('albums')` instead of `url.includes('albums?title')`. But I'm going to pass it now.

**Pull request merged**